Heavy Duty Stow Boards

DracoGriffin opened the pull request:
edit: If accepted, closes #23318

Utilized copy-from to reduce json bloat.

| | Durability| Weight (lbs) | Damage | Component Required | Cargo<br>Formula: ``(250*size)`` | Mechanic Skill Required
--- | :---: | :---: | :---: | :---: | :---: | :---:
Board | 240| 2.2| 100%| Sheet Metal| 0 |1
Heavy Duty Board | ``(240*4.167)=`` 1000| 28.4| 100%| Steel Plating| 0 |1
Stow Board | ``(240*0.833)=`` 200| 2.2| 100%| Sheet Metal| ``(250*250)=``<br> 62500 |2
Heavy Duty Stow Board | ``(200*4.167)=`` 830| 28.4| 100%| Steel Plating| ``(250*(0.76*250))=`` 47500 |2

I altered the Heavy Duty Stow Board based on the ratios listed above. Heavy Duty Board durability is the durability of Board times 4.167. Likewise, Stow Board durability is approximately 0.833 of Board's durability. Thus, Heavy Duty Stow Board durability should approximate to be 4.167 of Stow Board's durability.

Secondly, weight is based on the component item (in this case, Board and Stow Board both require sheet metal, which gives the 2.2 lbs and cannot be altered directly without making pseudo-items -- Heavy Duty Board and Heavy Duty Stow Board both require sheet plating).

Lastly, the formula for cargo is a flat 250 times the size specified by the vehicle part (for instance, Stow Board size is 250, 250*250 is equal to 62500). At first I used 0.833 between the durabilities of Heavy Duty Board and Heavy Duty Stow Board, but felt that the resultant cargo amount (52080~) was just a bit too high in relation to the Stow Board's cargo capacity. I lowered to about 0.75 but that produces a strange number in relation to cargo capacities (46875), so I bumped it to 0.76 which produces 47500.

Asmageddon commented:
The numbers sound extremely similar to mine, although I'd round durability from 830 to 800, and perhaps cargo from 47.5k to 45k, just to keep the numbers nice and round.

And I still think that cornerpiece variants, as per [#23318](https://github.com/CleverRaven/Cataclysm-DDA/issues/23318), should also be included, as they are a visual-only addition and stowboards are the only boards missing them.

DracoGriffin commented:
Jenkins rebuild

DracoGriffin commented:
Jenkins, rebuild.

DracoGriffin commented:
> The numbers sound extremely similar to mine, although I'd round durability from 830 to 800, and perhaps cargo from 47.5k to 45k, just to keep the numbers nice and round.
And I still think that cornerpiece variants, as per #23318, should also be included, as they are a visual-only addition and stowboards are the only boards missing them.

Given my other PRs and experience with numbers, trying to fit them into a formula is a bit better than coming up with numbers.

Durability lowering to 800 could work, as heavy duty variants seem to range between 4.167*original_durability and 4*original_durability, which 200*4 would be 800 durability.

So for 45000 L cargo space, what's the reasoning for 180 / 250 = 0.72? I suppose I could've decreased instead of increased. Either way, the strange thing is how other heavy duty variants don't penalize cargo capacity, so it's a bit difficult to judge where it should be (or just be the same). Additionally, x7.5 or x.75 seems to be a bit more common than x5 or x.5 where x is a number.

| Vehicle Part| Part Size | Cargo Space <br>(in L) |
|------------------------------|------|-------|
| door | 10 | 2.5 |
| opaque door | 10 | 2.5 |
| table | 15 | 3.75 |
| wood table | 15 | 3.75 |
| reclining seat | 25 | 6.25 |
| storage bag | 60 | 15 |
| wire bike basket | 60 | 15 |
| seat | 100 | 25 |
| folding wood box | 120 | 30 |
| hatch | 150 | 37.5 |
| opaque hatch | 150 | 37.5 |
| heavy duty hatch | 150 | 37.5 |
| opaque heavy duty hatch | 150 | 37.5 |
| wood box | 150 | 37.5 |
| washing machine | 160 | 40 |
| bed | 200 | 50 |
| travois | 200 | 50 |
| shopping cart basket | 200 | 50 |
| folding shopping cart basket | 200 | 50 |
| kitchen | 200 | 50 |
| weld rig | 200 | 50 |
| foodco rig | 200 | 50 |
| chemlab | 200 | 50 |
| forge | 200 | 50 |
| stow board | 250 | 62.5 |
| box | 275 | 68.75 |
| minifridge | 300 | 75 |
| floor trunk | 350 | 87.5 |
| trunk | 650 | 162.5 |
| cargo space | 1000 | 250 |

And I'd be for adding corner pieces for stow boards, but if I only did that for the heavy duty variants, then corners would be needed for regular stow boards as well, which I'd add those as well, but not sure if that'd be allowed within this PR. @illi-kun suggestions?

Asmageddon commented:
My reasoning behind nerfing storage space of heavy duty stowboards is twofold: I imagine it would need considerably more structural elements to be durable enough, and secondly, 800hp is for most purposes easily sturdy enough, and if the storage isn't reduced, it'll essentially be a "straight upgrade" over heavy duty boards, if not normal stowboards as well.

DracoGriffin commented:
Added the new variants (heavy duty stow board) and corner variants (for both stow boards and heavy duty version).

Also, moved stow boards from json/vehicle_parts.json to json/vehicleparts/boards.json (since they are technically boards) and condensed with abstract/copyfrom/proportional.

Secondly, looks like I found some errors in other boards:
```json
  {
    "abstract": "cloth_halfboard",
    "copy-from": "clothboard",
    "type": "vehicle_part",
    "name": "cloth quarterpanel",
    **"relative": { "durability": 0.8 },**
    "delete": { "flags": [ "OPAQUE" ] }
  },
  {
    "abstract": "halfboard",
    "copy-from": "board",
    "type": "vehicle_part",
    "name": "quarterpanel",
    **"relative": { "durability": 0.8 },**
    "delete": { "flags": [ "OPAQUE" ] }
  },
  {
    "abstract": "hdhalfboard",
    "copy-from": "hdboard",
    "type": "vehicle_part",
    "name": "heavy duty quarterpanel",
    **"relative": { "durability": 0.8 },**
    "delete": { "flags": [ "OPAQUE" ] }
  },
  {
    "abstract": "woodhalfboard",
    "copy-from": "woodboard",
    "type": "vehicle_part",
    "name": "wooden quarterpanel",
    **"relative": { "durability": 0.8 },**
    "delete": { "flags": [ "OPAQUE" ] }
  },
```

So if I'm understanding correctly, relative justs add a flat number onto the called source, so adding 0.8 durability makes no sense (240 -> 240.8), so I'm assuming this is actually meant to be `"proportional" : { "durability" : 0.8 }` (for 240 -> 192). I'll open a separate PR for that fix, however, that may end up influencing a new durability for stow boards since quarterpanels will now be 192 durability instead of the bugged 240.

DracoGriffin commented:
Didn't have any issues when testing with changes, but checks were still failing. @ZhilkinSerg suggested to linting in Discord, so testing to see if that resolves failed checks. Additionally, updated the entries from most current master to hopefully resolve those as well.

ZhilkinSerg commented on data/json/vehicle_parts.json:
```diff
@@ -1,2503 +1,2215 @@
-          },
-          "removal": {
-            "skills": [ [ "mechanics", 1 ] ],
-            "qualities": [ { "id": "SCREW", "level": 1 } 
```
Newline at end of file is missing.

ZhilkinSerg commented on data/json/vehicleparts/boards.json:
```diff
@@ -475,4 +572,4 @@
     "symbol": "H"
   }
-]
+]
```
Also newline at end of file is missing.

DracoGriffin replied:
Fixed.

ZhilkinSerg commented on data/json/vehicle_parts.json:
```diff
@@ -1,32 +1,191 @@
+      "install": { "time": 20000, "skills": [ [ "mechanics", 2 ] ], "qualities": [ { "id": "WRENCH", "level": 2 } ] },
+      "repair": { "skills": [ [ "mechanics", 2 ] ], "time": 2000, "using": [ [ "welding_standard", 5 ] ] },
+      "removal": { "skills": [ [ "mechanics", 2 ] ], "qualities": [ { "id": "WRENCH", "level": 2 } ] }
     },
```
`}` is missing before `},`

ZhilkinSerg commented on data/json/vehicle_parts.json:
```diff
@@ -1893,592 +2052,1863 @@
-            {"item": "wire", "count": [4, 8]},
-            {"item": "plastic_chunk", "count": [4, 8]}
-        ]
+    "flags": [ "ROOF", "FOLDABLE" ],
```
Broken syntax here and in a couple of lines below.

DracoGriffin replied:
Yeah, something went weird on that conflict merge. I'm tempted to just close this PR and re-open it; the commits are way too messy. It was giving me issues with astyle, but when I fixed that, it made vehicle_parts.json have a huge diff change.
